sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuTransitionOverrides.scala

```diff
@@ -396,13 +397,17 @@ class GpuTransitionOverrides extends Rule[SparkPlan] {
       // Insert a sort after the last hash-based op before the query result if there are no
       // intermediate nodes that have a specified sort order. This helps with the size of
       // Parquet and Orc files
+      // FIXME: This is passing a GPU sort order as a CPU sort order since there is not an easy
```

Comment:
is this something you plan to tackle here? or a subsequent issue. would we need to `columnarEval` and turn it into a cpu literal? I assume that you can't have a column vector here (just thinking out loud)

Reply:
On the same note here, should we require/assert that that the order is not a complex expression?

Reply:
This isn't a runtime need-to-execute-this problem, rather a plans-match-up problem.  Note that before this PR, we're reporting potentially a GPU SortOrder expression and it isn't going to fail.  The worst that should happen is that the planner doesn't realize that some CPU nodes later in the plan that require a certain sort order already lines up with what this node is doing because CPU and GPU expressions won't semantically compare unless they're trivial attribute references.  If it doesn't compare then it will insert a redundant sort to "fix" the required ordering.

However I just noticed that this code is explicitly checking for the case where no node later in the plan specify a sort order.  Therefore this should never be a problem in practice, since nothing should care about the advertised sort order expression.  I'll update the comment to say we're cheating by advertising GPU sort expressions but explain why that's probably OK.